packages/flutter/lib/src/rendering/list_wheel_viewport.dart

```diff
@@ -529,35 +553,169 @@ class RenderListWheelViewport
+        _destroyChild(firstChild);
+      return;
+    }
+
+    // Now there are 2 cases:
+    //  - The target child list and our current child list have intersection:
+    //    We shorten and extend our current child list so that the two lists
+    //    match. Most of the time we are in this case.
+    //  - The target list and our current child list have no intersection:
+    //    We first remove all children and then add one child from the target
+    //    list => this case becomes the other case.
+
+    // Case when there is no intersection.
+    if (childCount > 0 &&
+        (indexOf(firstChild) > targetLastIndex || indexOf(lastChild) < targetFirstIndex)) {
+      while (firstChild != null)
+        _destroyChild(firstChild);
+    }
+
+    // If there is no child at this stage, we add the first one that is in
+    // target range.
+    if (childCount == 0) {
+      _createChild(targetFirstIndex);
+      _layoutChild(firstChild, childConstraints, targetFirstIndex);
+    }
+
+    int currentFirstIndex = indexOf(firstChild);
+    int currentLastIndex = indexOf(lastChild);
+
+    // Remove all unnecessary children by shortening the current child list, in
+    // both directions.
+    while (currentFirstIndex < targetFirstIndex) {
+      _destroyChild(firstChild);
+      currentFirstIndex++;
+    }
+    while (currentLastIndex > targetLastIndex) {
+      _destroyChild(lastChild);
+      currentLastIndex--;
+    }
+
+    // Relayout all active children.
     RenderBox child = firstChild;
-    final BoxConstraints innerConstraints =
-        constraints.copyWith(
-          minHeight: _itemExtent,
-          maxHeight: _itemExtent,
-          minWidth: 0.0,
-        );
     while (child != null) {
-      child.layout(innerConstraints, parentUsesSize: true);
-      final ListWheelParentData childParentData = child.parentData;
-      // Centers the child in the cross axis. Consider making it configurable.
-      final double crossPosition = size.width / 2.0 - child.size.width / 2.0;
-      childParentData.offset = new Offset(crossPosition, currentOffset);
-      currentOffset += _itemExtent;
+      child.layout(childConstraints, parentUsesSize: true);
       child = childAfter(child);
     }
 
+    // Spawning new children that are actually visible but not in child list yet.
+    while (currentFirstIndex > targetFirstIndex) {
+      _createChild(currentFirstIndex - 1);
+      _layoutChild(firstChild, childConstraints, --currentFirstIndex);
+    }
+    while (currentLastIndex < targetLastIndex) {
+      _createChild(currentLastIndex + 1, after: lastChild);
+      _layoutChild(lastChild, childConstraints, ++currentLastIndex);
+    }
+
     offset.applyViewportDimension(_viewportExtent);
-    offset.applyContentDimensions(_minScrollExtent, _maxScrollExtent);
+
+    // Applying content dimensions bases on how the childManager builds widgets:
+    // if it is available to provide a child just out of target range, then
```

Comment:
What does this mean?

Reply:
applyContentDimensions() will set the limit to how much user can scroll.
Since the number of children can be unknown, I reset the limit every time performLayout() is called. Details are in the comments.